source/Junkership.js

```diff
@@ -4,69 +4,74 @@ var Keyb = require("keyb")
 import Reference from "./Reference.js"
 import Projectile from "./Projectile.js"
 import Score from "./Score.js"
-import {PeaShoota, PowerUp, TriShoota, FiveShoota, RapidFire, SprayShot,SuperSprayShot, CrazySprayShot, VertSprayShot, VertShoota, RapidSprayShot, BFG} from "./PowerUp.js"
-
+import {PeaShoota, PowerUp, TriShoota, FiveShoota, RapidFire, SprayShot,
+        SuperSprayShot, CrazySprayShot, VertSprayShot, VertShoota,
+        RapidSprayShot, BFG} from "./PowerUp.js"
 
 export default class Junkership extends Pixi.Sprite {
-    constructor() {
-        super(PIXI.loader.resources.redJunkership.texture)
+    constructor(controlScheme) {
+        super(checkTex())
         game.playerCount++
         this.speed = 60
         this.score = new Score()
         this.powerUp = new PeaShoota()
         this.reloadTime = 0
+        this.controls = controlScheme
         this.hitBox = new Pixi.Rectangle(
             this.x + 1 , // Left offset
             this.y + 1 , // Top offset
             this.width - 3 , // Right offset + left offset
             this.height - 3 )// Bottom offset + top offset
-        this.WeaponList = [PeaShoota, TriShoota, FiveShoota, RapidFire, SprayShot, SuperSprayShot, CrazySprayShot, VertSprayShot, VertShoota, RapidSprayShot, BFG]
+        this.WeaponList = [PeaShoota, TriShoota, FiveShoota, RapidFire,
+                           SprayShot, SuperSprayShot, CrazySprayShot,
+                           VertSprayShot, VertShoota, RapidSprayShot, BFG]
     }
-    
-    update(delta) {
-        // Ugly kludge
-        if (this.width === 1) {
-            this.onCollision()
-        }
 
+    update(delta) {
         var relativeSpeed = this.speed * delta
 
-        if (Keyb.isJustDown("<up>")) {
+        if (Keyb.isJustDown(Reference.ControlScheme.keys[this.controls].up)) {
```

Comment:
Done. Still passes an index to the constructor, but uses that to pull in `Reference.ControlScheme.keys[index]`, letting the controls call as you mentioned.